be/src/vec/functions/function_truncate.h

```diff
@@ -1,245 +0,0 @@
-    // should NOT behave like two column arguments, so we can not use const column default implementation
-    bool use_default_implementation_for_constants() const override { return false; }
-
-    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
-                        size_t result, size_t input_rows_count) const override {
-        const ColumnWithTypeAndName& column_general = block.get_by_position(arguments[0]);
-        ColumnPtr res;
-
-        // potential argument types:
-        // 0. truncate(ColumnConst, ColumnConst)
-        // 1. truncate(Column), truncate(Column, ColumnConst)
-        // 2. truncate(Column, Column)
-        // 3. truncate(ColumnConst, Column)
-
-        if (arguments.size() == 2 && is_column_const(*block.get_by_position(arguments[0]).column) &&
-            is_column_const(*block.get_by_position(arguments[1]).column)) {
-            // truncate(ColumnConst, ColumnConst)
-            auto col_general =
-                    assert_cast<const ColumnConst&>(*column_general.column).get_data_column_ptr();
-            Int16 scale_arg = 0;
-            RETURN_IF_ERROR(FunctionTruncate<Impl>::get_scale_arg(
-                    block.get_by_position(arguments[1]), &scale_arg));
-
-            auto call = [&](const auto& types) -> bool {
-                using Types = std::decay_t<decltype(types)>;
-                using DataType = typename Types::LeftType;
-
-                if constexpr (IsDataTypeNumber<DataType> || IsDataTypeDecimal<DataType>) {
-                    using FieldType = typename DataType::FieldType;
-                    res = Dispatcher<FieldType, RoundingMode::Trunc,
-                                     TieBreakingMode::Auto>::apply_vec_const(col_general,
-                                                                             scale_arg);
-                    return true;
-                }
-
-                return false;
-            };
-
-#if !defined(__SSE4_1__) && !defined(__aarch64__)
-            /// In case of "nearbyint" function is used, we should ensure the expected rounding mode for the Banker's rounding.
-            /// Actually it is by default. But we will set it just in case.
-
-            if constexpr (rounding_mode == RoundingMode::Round) {
-                if (0 != fesetround(FE_TONEAREST)) {
-                    return Status::InvalidArgument("Cannot set floating point rounding mode");
-                }
-            }
-#endif
-
-            if (!call_on_index_and_data_type<void>(column_general.type->get_type_id(), call)) {
-                return Status::InvalidArgument("Invalid argument type {} for function {}",
-                                               column_general.type->get_name(), "truncate");
-            }
-            // Important, make sure the result column has the same size as the input column
-            res = ColumnConst::create(std::move(res), input_rows_count);
```

Comment:
This logical should be reserved.

Reply:
If constant implement is false, indeed it is necessary.